src/mapblock.h

```diff
@@ -280,6 +287,10 @@ class MapBlock
 
 	inline void setNodeNoCheck(s16 x, s16 y, s16 z, MapNode n)
```

Comment:
we should investigate the callers of `setNode*` to make sure they don't call setNode when the wanted node is already there

Reply:
I considered that, but then it seemed overkill. Happy to add that little bit of code.

Reply:
I wouldn't add it here, but rather to the call sites. If the call sites are not suitable for such a check (e.g. nested loop) I wouldn't add it at all.

Reply:
Then I wouldn't add it at all. Putting this on the callsites complicates the code.